### PR TITLE
💄 Add input for collector's message

### PR DESCRIPTION
**Pull request by AuroraHuang22**
截圖 2023-01-11 下午5 41 53

截圖 2023-01-11 下午5 42 02

### CODE REVIEW

File: src/locales/en.json [hunk 1]
```diff
@@ -240,6 +240,8 @@
   "nft_class_details_info_section_nft_class_id_label": "NFT Class ID",
   "nft_class_details_info_section_title": "Info",
   "nft_class_uncollectible": "Uncollectible",
+  "nft_collect_modal_leave_message": "Leave message with creator",
```
Comment (williamchong) [inline src/locales/en.json:243]:
```suggestion
  "nft_collect_modal_leave_message": "Leave a message to creator",

```

File: src/components/EventModal/Collect/index.vue [hunk 2]
```diff
@@ -4,7 +4,10 @@
     :has-close-button="isShowCloseButton"
     :header-text="headerText"
     preset="collect"
-    @close="$emit('close')"
+    @close="
+      () => {
+        $emit('close'), (shouldShowMessageInput = false);
+      }"
```
Comment (nwingt) [inline src/components/EventModal/Collect/index.vue:10]: Better write in method

File: src/components/EventModal/Collect/index.vue [hunk 3]
```diff
@@ -160,6 +196,8 @@ export default {
     return {
       paymentMethod: undefined,
       justCollectedNFTId: undefined,
+      shouldShowMessageInput: false,
+      memo: undefined,
```
Comment (nwingt) [inline src/components/EventModal/Collect/index.vue:200]:
```suggestion
      memo: '',
```

File: src/components/EventModal/Collect/index.vue [hunk 4]
```diff
@@ -225,6 +263,9 @@ export default {
     if (this.classId) {
       this.resetState();
     }
+    if (window.sessionStorage) {
+      this.memo = window.sessionStorage.getItem('COLLECT_MESSAGE');
```
Comment (nwingt) [inline src/components/EventModal/Collect/index.vue:267]:
```suggestion
      this.memo = window.sessionStorage.getItem('COLLECTOR_MESSAGE');
```

File: src/locales/en.json (hunk 1 above)
Comment (nwingt) [inline src/locales/en.json:243]:
```suggestion
  "nft_collect_modal_leave_message": "Leave message for creator",

```

File: src/components/EventModal/Collect/index.vue (hunk 4 above)
Comment (williamchong) [inline src/components/EventModal/Collect/index.vue:267]: Need to handle clear on different class id

**Review by williamchong — CHANGES_REQUESTED** [review]

**Review by nwingt — COMMENTED** [review]
Please add event

**Review by nwingt — APPROVED** [review]

**Review by williamchong — APPROVED** [review]

**Pull request merged**